apps/frontend/src/libs/components/header/header.tsx

```diff
@@ -26,11 +37,31 @@ const Header = (): JSX.Element => {
 		({ auth }) => auth.authenticatedUser,
 	);
 
+	const { unreadNotifications } = useAppSelector(
+		({ notifications }) => notifications,
+	);
+
+	useEffect(() => {
+		void dispatch(notificationActions.loadAllUnread());
+	}, [dispatch]);
+
+	const markAllNotificationsAsRead = useCallback(() => {
+		for (const notification of unreadNotifications) {
+			void dispatch(notificationActions.markAsRead({ id: notification.id }));
+		}
+	}, [dispatch, unreadNotifications]);
```

Comment:
This will be very inefficient. Instead let's use /read route to mark all notifications as read at once, and handle them as a bulk on backend side too.